src/write-bin-template/write-file-to-bin.ts

```diff
@@ -0,0 +1,34 @@
+import { makeDirectorySafelyFactory } from './make-directory-safely'
+import { getTemplateFileStringsFactory } from './get-template-file-strings'
+import { getBinTargetPathsFactory } from './get-bin-target-paths'
+import { FileSystem, Path } from './types'
+
+export type WriteFilesToBin = (jsonPath: string) => void
+
+export const writeFileToBinFactory = (
+  fileSystem: FileSystem,
+  path: Path,
+  {
+    sourcePathFragment,
+    targetPathFragment,
+  }: {
+    sourcePathFragment: string
+    targetPathFragment: string
+  },
+): WriteFilesToBin => {
+  const getBinTargetPaths = getBinTargetPathsFactory({
+    sourcePathFragment,
+    targetPathFragment,
+  })
+
+  return (jsonPath: string): void => {
+    const { jsonFileStr, yamlFileStr } =
+      getTemplateFileStringsFactory(fileSystem)(jsonPath)
+    const { jsonBinPath, yamlBinPath } = getBinTargetPaths(jsonPath)
+    const makeDirectory = makeDirectorySafelyFactory(fileSystem)
+    makeDirectory(path.dirname(jsonBinPath))
```

Comment:
Not sure if this is easier to read or not, but does reduce duplication 🤷 

```suggestion
    [
      [jsonBinPath, jsonFileStr],
      [yamlBinPath, yamlFileStr],
    ].forEach(([binPath, fileContent]) => {
      makeDirectory(path.dirname(binPath))
      fileSystem.writeFileSync(binPath, fileContent)
    })
```